File: server/mb-platform/platform-iam/src/main/java/com/metabuild/platform/iam/api/AuthApi.java
package com.metabuild.platform.iam.api;

import com.metabuild.common.security.LoginResult;
import com.metabuild.platform.iam.api.cmd.LoginCmd;

/**
 * 认证模块对外 API 接口（供跨模块调用）。
 */
public interface AuthApi {

    LoginResult login(LoginCmd request);

    void logout();
}

File: client/packages/api-sdk/src/apis/auth-api.ts
import { getClient } from '../config';
import type { LoginCommand, LoginView, CurrentUserView } from '../types/auth';

export const authApi = {
  login(cmd: LoginCommand): Promise<LoginView> {
    return getClient().request('/api/v1/auth/login', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(cmd),
    });
  },

  logout(): Promise<void> {
    return getClient().request('/api/v1/auth/logout', { method: 'POST' });
  },

  refresh(refreshToken: string): Promise<LoginView> {
    return getClient().request('/api/v1/auth/refresh', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ refreshToken }),
    });
  },

  /** 获取当前登录用户信息（GET /auth/me，需认证） */
  getCurrentUser(): Promise<CurrentUserView> {
    return getClient().request('/api/v1/auth/me');
  },
};
